tests/components/squeezebox/test_media_browser.py

```
@@ -0,0 +1,193 @@
+                "media_content_type": "Tracks",
+            }
+        )
+        response = await client.receive_json()
+        assert response["success"]
+        tracks = response["result"]
+        assert tracks["title"] == "titles"
+        assert len(tracks["children"]) == 3
+
+
+async def test_async_browse_error(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Search for a non-existent item and assert error."""
+    client = await hass_ws_client()
+    await client.send_json(
+        {
+            "id": 1,
+            "type": "media_player/browse_media",
+            "entity_id": "media_player.test_player",
+            "media_content_id": "0",
+            "media_content_type": MediaType.ALBUM,
+        }
+    )
+    response = await client.receive_json()
+    assert not response["success"]
+
+
+async def test_play_browse_item(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test play browse item."""
+    await hass.services.async_call(
+        "media_player",
+        "play_media",
+        {
+            "entity_id": "media_player.test_player",
+            "media_content_id": "1234",
+            "media_content_type": "album",
+        },
+    )
+
+
+async def test_play_browse_item_nonexistent(
+    hass: HomeAssistant, config_entry: MockConfigEntry
+) -> None:
+    """Test trying to play an item that doesn't exist."""
+    await hass.services.async_call(
+        "media_player",
+        "play_media",
+        {
+            "entity_id": "media_player.test_player",
+            "media_content_id": "0",
+            "media_content_type": "album",
+        },
+    )
+
+
+async def test_play_browse_item_bad_category(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test trying to play an item whose category doesn't exist."""
+    await hass.services.async_call(
+        "media_player",
```

What should happen when we call this? Does this raise something?

---

Good question. I added this just for code coverage (right now it fails silently, but looking at
other integrations like `sonos` they raise a BrowseError exception). I'm fixing that now.